packages/manager/apps/ips/src/data/hooks/useExportToFile.ts

```diff
@@ -0,0 +1,92 @@
+import { useState, useContext } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useNotifications } from '@ovh-ux/manager-react-components';
+import { aapi } from '@ovh-ux/manager-core-api';
+import { mkConfig, generateCsv, download } from 'export-to-csv';
+import { ListingContext } from '@/pages/listing/listingContext';
+import { ExportIpToCsvData } from '@/data/api';
+import { TRANSLATION_NAMESPACES } from '@/utils';
+
+const exportColumns = [
+  {
+    label: 'IP Block',
+    getValue: (item: ExportIpToCsvData) => item.ipBlock,
+  },
+  {
+    label: 'Version',
+    getValue: (item: ExportIpToCsvData) => item.version,
+  },
+  {
+    label: 'Type',
+    getValue: (item: ExportIpToCsvData) => item.type,
+  },
+  {
+    label: 'Country',
+    getValue: (item: ExportIpToCsvData) => item.country || '',
+  },
+  {
+    label: 'Service Name',
+    getValue: (item: ExportIpToCsvData) => item.routedTo?.serviceName || '',
+  },
+  {
+    label: 'Description',
+    getValue: (item: ExportIpToCsvData) => item.description || '',
+  },
+];
+
+export const useExportToCsv = () => {
+  const { t } = useTranslation([TRANSLATION_NAMESPACES.exportIpToCsv]);
+  const { apiFilter, ipToSearch } = useContext(ListingContext);
+  const { addSuccess, addError } = useNotifications();
+  const [isCSVLoading, setIsCSVLoading] = useState(false);
+
+  const handleExportToCsv = async () => {
+    setIsCSVLoading(true);
+    try {
+      const result = await aapi
+        .get('/ipsexportcsv', {
+          params: {
+            ...apiFilter,
+            serviceName: apiFilter['routedTo.serviceName'],
+            ip: ipToSearch,
+          },
+        })
+        .then(({ data }) => data);
+
+      if (result && result?.data && result?.data.length > 0) {
+        const csvConfig = mkConfig({
+          filename: 'export_ips',
+          fieldSeparator: ',',
+          quoteStrings: true,
+          decimalSeparator: '.',
+          useKeysAsHeaders: true,
+        });
+
+        const csvData = result.data.map((item: ExportIpToCsvData) =>
+          exportColumns.reduce(
+            (acc, column) => ({
+              ...acc,
+              [column.label]: column.getValue(item),
+            }),
+            {},
+          ),
+        );
```

Comment:
It feels complex compute for what it does. Maybe we can simplify it and also remove the reduce inside the map :arrow_down: 

```suggestion
        const csvData = result.data.map((item: ExportIpToCsvData) =>
          ({
            'IP Block': item.ipBlock,
            'Version': item.version,
            'Type': item.type,
            'Country': item.country || '',
            'Service Name': item.routedTo?.serviceName || '',
            'Description': item.description || ''
          )}
        );
```